From 04347c3d46e429a50de4ce2286fed458d6dbb7b8 Mon Sep 17 00:00:00 2001
From: RubenKelevra <ruben@vfn-nrw.de>
Date: Thu, 22 Jun 2017 00:20:28 +0200
Subject: [PATCH] deny all ipv6 input from client-network as well as dhcpv6 /
 RA delivery / RS reception we currently doesn't use IPv6 at all, so we
 disable those unnecessary overhead until we use ipv6 in the networks

---
 .../files/lib/gluon/ebtables/200-dir-dhcpv6              |  5 -----
 .../files/lib/gluon/ebtables/200-dir-radv                | 16 ++++++++++++----
 .../lib/gluon/ebtables/110-local-forward-allow-ipv6      |  8 ++++----
 .../files/lib/gluon/ebtables/100-dir-chain               |  4 ++++
 .../files/lib/gluon/ebtables/101-dir-rules               | 12 ++++++++++++
 5 files changed, 32 insertions(+), 13 deletions(-)
 delete mode 100644 package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-dhcpv6

diff --git a/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-dhcpv6 b/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-dhcpv6
deleted file mode 100644
index 470a7648..00000000
--- a/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-dhcpv6
+++ /dev/null
@@ -1,5 +0,0 @@
-rule 'FORWARD -p IPv6 --ip6-protocol udp --ip6-destination-port 547 -j OUT_ONLY'
-rule 'OUTPUT -p IPv6 --ip6-protocol udp --ip6-destination-port 547 -j OUT_ONLY'
-
-rule 'FORWARD -p IPv6 --ip6-protocol udp --ip6-destination-port 546 -j IN_ONLY'
-rule 'INPUT -p IPv6 --ip6-protocol udp --ip6-destination-port 546 -j IN_ONLY'
diff --git a/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-radv b/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-radv
index b34d4c76..f1307b8e 100644
--- a/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-radv
+++ b/package/gluon-ebtables-filter-ra-dhcp/files/lib/gluon/ebtables/200-dir-radv
@@ -1,5 +1,13 @@
-rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j OUT_ONLY'
-rule 'OUTPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j OUT_ONLY'
+rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j OUT_ONLY_BAT'
+rule 'OUTPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j OUT_ONLY_BAT'
+
+rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisment -j OUT_ONLY_BAT'
+rule 'OUTPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisment -j OUT_ONLY_BAT'
+
+rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisement -j IN_ONLY_LOCAL'
+rule 'INPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisement -j IN_ONLY_LOCAL'
+
+rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j IN_ONLY_BAT'
+rule 'INPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-solicitation -j IN_ONLY_BAT'
+
 
-rule 'FORWARD -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisement -j IN_ONLY'
-rule 'INPUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type router-advertisement -j IN_ONLY'
diff --git a/package/gluon-ebtables-source-filter/files/lib/gluon/ebtables/110-local-forward-allow-ipv6 b/package/gluon-ebtables-source-filter/files/lib/gluon/ebtables/110-local-forward-allow-ipv6
index 9b2d1147..6d818a5f 100644
--- a/package/gluon-ebtables-source-filter/files/lib/gluon/ebtables/110-local-forward-allow-ipv6
+++ b/package/gluon-ebtables-source-filter/files/lib/gluon/ebtables/110-local-forward-allow-ipv6
@@ -1,9 +1,9 @@
 site = require('gluon.site_config')
 
-rule('LOCAL_FORWARD -p IPv6 --ip6-src fe80::/64 -j RETURN')
-rule('LOCAL_FORWARD -p IPv6 --ip6-src ::/128 --ip6-proto ipv6-icmp -j RETURN')
-rule('LOCAL_FORWARD -p IPv6 --ip6-src ' .. site.prefix6 .. ' -j RETURN')
+rule('LOCAL_FORWARD -p IPv6 --logical-in br-client -i local-port --ip6-src fe80::/64 -j RETURN')
+rule('LOCAL_FORWARD -p IPv6 --logical-in br-client -i local-port --ip6-src ::/128 --ip6-proto ipv6-icmp -j RETURN')
+rule('LOCAL_FORWARD -p IPv6 --logical-in br-client -i local-port --ip6-src ' .. site.prefix6 .. ' -j RETURN')
 
 for _, prefix in ipairs(site.extra_prefixes6 or {}) do
-	rule('LOCAL_FORWARD -p IPv6 --ip6-src ' .. prefix .. ' -j RETURN')
+	rule('LOCAL_FORWARD -p IPv6 --logical-in br-client -i local-port --ip6-src ' .. prefix .. ' -j RETURN')
 end
diff --git a/package/gluon-ebtables/files/lib/gluon/ebtables/100-dir-chain b/package/gluon-ebtables/files/lib/gluon/ebtables/100-dir-chain
index e6bf98e3..dcbe221a 100644
--- a/package/gluon-ebtables/files/lib/gluon/ebtables/100-dir-chain
+++ b/package/gluon-ebtables/files/lib/gluon/ebtables/100-dir-chain
@@ -1,5 +1,9 @@
 chain('IN_ONLY', 'RETURN')
 chain('OUT_ONLY', 'RETURN')
+chain('IN_ONLY_BAT', 'RETURN')
+chain('OUT_ONLY_BAT', 'RETURN')
+chain('IN_ONLY_LOCAL', 'RETURN')
+chain('OUT_ONLY_LOCAL', 'RETURN')
 
 chain('MULTICAST_OUT', 'RETURN')
 chain('MULTICAST_OUT_ICMPV6', 'RETURN')
diff --git a/package/gluon-ebtables/files/lib/gluon/ebtables/101-dir-rules b/package/gluon-ebtables/files/lib/gluon/ebtables/101-dir-rules
index 74486ae5..4a7a244a 100644
--- a/package/gluon-ebtables/files/lib/gluon/ebtables/101-dir-rules
+++ b/package/gluon-ebtables/files/lib/gluon/ebtables/101-dir-rules
@@ -2,6 +2,18 @@ rule 'IN_ONLY --logical-in br-client -i bat0 -j RETURN'
 rule 'IN_ONLY --logical-in br-client -i local-port -j RETURN'
 rule 'IN_ONLY --logical-in br-client -j DROP'
 
+rule 'IN_ONLY_BAT --logical-in br-client -i bat0 -j RETURN'
+rule 'IN_ONLY_BAT --logical-in br-client -j DROP'
+
+rule 'IN_ONLY_LOCAL --logical-in br-client -i local-port -j RETURN'
+rule 'IN_ONLY_LOCAL --logical-in br-client -j DROP'
+
 rule 'OUT_ONLY --logical-out br-client -o bat0 -j RETURN'
 rule 'OUT_ONLY --logical-out br-client -o local-port -j RETURN'
 rule 'OUT_ONLY --logical-out br-client -j DROP'
+
+rule 'OUT_ONLY_BAT --logical-out br-client -o bat0 -j RETURN'
+rule 'OUT_ONLY_BAT --logical-out br-client -j DROP'
+
+rule 'OUT_ONLY_LOCAL --logical-out br-client -o local-port -j RETURN'
+rule 'OUT_ONLY_LOCAL --logical-out br-client -j DROP'
-- 
2.13.1

